### DISCUSSION TITLE
3D Geology Model(Table top view)

### DISCUSSION THREAD

**Discussion by Kavita-lab** [pyvista/pyvista#4436] [Q&A]
Hello Sir,
Hope you are doing well.
I am writing with regards to a feature implementation in pyvista.
I need implement a table top view after wrapping a topographic map over DEM.
Please refer the below mentioned link and image:
link:    https://docs.pyvista.org/version/stable/examples/02-plot/topo-map.html#topographic-map
Kindly  help me to create the 3D geology model (table top view) of the topographic surface so that I can visualize dense base along with elevation.

![labelled screenshot](https://github.com/pyvista/pyvista/assets/73441674/5fb34275-31bd-474a-8250-ef3ba1c4eaa2)

**Comment by banesullivan** [comment]
You probably want to combine that example with the "extrude trim" example: https://docs.pyvista.org/version/stable/examples/01-filter/extrude-trim.html#extrude-trim

Here is a start, but there are issues with texture mapping the extruded surface:

```py
import pyvista as pv
from pyvista import examples

# Load the elevation data as a surface
elevation = examples.download_crater_topo().warp_by_scalar()
# Load the topographic map from a GeoTiff
topo_map = examples.download_crater_imagery()
topo_map = topo_map.flip_y()  # flip to align to our dataset

# Bounds of the aerial imagery - given to us
bounds = (1818000, 1824500, 5645000, 5652500, 0, 3000)
# Clip the elevation dataset to the map's extent
local = elevation.clip_box(bounds, invert=False)
# Apply texturing coordinates to associate the image to the surface
local.texture_map_to_plane(use_bounds=True, inplace=True)

n = 30
plane = pv.Plane(
    center=(local.center[0], local.center[1], -1),
    direction=(0, 0, -1),
    i_size=local.length,
    j_size=local.length,
)

# Perform the extrude trim
extruded = local.extract_surface().extrude_trim((0, 0, -1.0), plane)

pl = pv.Plotter()
# pl.add_mesh(local, texture=topo_map)
pl.add_mesh(extruded, color=True)
pl.enable_terrain_style()
pl.show()
```

![screenshot (4)](https://github.com/pyvista/pyvista/assets/22067021/0c563fce-0da1-4406-8210-9ed337551f66)
